webapp/packages/core-utils/src/ILoadableState.test.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { getFirstException, ILoadableState, isContainsException, isLoadableStateHasException } from './ILoadableState';
+
+const getMockedLoadableState = (state: Partial<ILoadableState>): ILoadableState => ({
+  exception: new Error(),
+  isLoading: () => false,
+  isLoaded: () => false,
+  isError: () => false,
+  load: () => undefined,
```

Comment:
this doesn't seems to be correct mock, because isError always returns false, but you specified default exception value

probably you want to implement load function that accepts callback and catches excetions and for isError you need to use isContainsException

Reply:
I think you a bit overcomplicating here

We dont need to test integration between functions, it not quite a unit test apporach

Changed `getMockedLoadableState` with null exception. + in test cases added `isError` parameter just in case so the mock is more truthful

But since it is (`isError`) not tested yet it is not important. But we can add some logic here if it is important since cause you mentioned it.

The most important moment here we should implement new rule in our team:
If you have changed helper's/component's/etc logic. You have to also support this new logic with unit tests written for this particular file

Reply:
true